tests/test_tracing.py

```diff
@@ -151,7 +154,31 @@ def test_span_trimming(sentry_init, capture_events):
 
 
 def test_nested_span_sampling_override():
-    with Hub.current.start_span(transaction="outer", sampled=True) as span:
+    with start_transaction(name="outer", sampled=True) as span:
+        assert span.sampled is True
+        with start_transaction(name="inner", sampled=False) as span2:
+            assert span2.sampled is False
+
         assert span.sampled is True
-        with Hub.current.start_span(transaction="inner", sampled=False) as span:
-            assert span.sampled is False
+
+
+def test_transaction_method_signature(sentry_init, capture_events):
+    sentry_init(traces_sample_rate=1.0)
+    events = capture_events()
+
+    with pytest.raises(TypeError):
+        start_span(name="foo")
+
+    with pytest.raises(ValueError):
+        start_transaction()
+
+    with start_transaction(name="a"):
+        pass
+
+    with start_transaction("b"):
+        pass
+
+    with start_transaction(Transaction(name="c")):
+        pass
```

Comment:
This takes a transaction object because `continue_from_headers` is in use in Python. In JS the workaround seems to be to take the return value of `Transaction.continue_from_headers`, extract all relevant properties (span/trace ID) and pass them back as context (which in python would map to kwargs) into `start_transaction`. I don't think that's a good API

Reply:
My current proposal here is `with Transaction(name="c"):`, and `Transaction` and `Span` would both be context managers.

With that, neither `Hub.start_transaction` nor `start_transaction` would be necessary.

But maybe we won't land it in this PR.